packages/api/src/mappings/tenovi/body.ts

```diff
@@ -6,31 +6,39 @@ import { getFloatValue, formatNumber } from "../../shared/numbers";
 import convert from "convert-units";
 
 export const mapToBody = (date: string, tenoviWeight: TenoviMeasurementData): Body => {
-  const { hwi_device_id: id, device_name: name, sensor_code } = tenoviWeight[0];
-  const metadata = {
-    date: date,
-    source: PROVIDER_TENOVI,
-    data_source: {
-      source_type: SourceType.device,
-      id,
-      name,
-      type: sensor_code,
+  const body: Body = {
+    metadata: {
+      date: date,
+      source: PROVIDER_TENOVI,
     },
   };
 
-  const body: Body = {
-    metadata,
-  };
+  if (tenoviWeight.length) {
+    const { hwi_device_id: id, device_name: name, sensor_code } = tenoviWeight[0];
+    const sourceInfo = {
+      data_source: {
+        source_type: SourceType.device,
+        id,
+        name,
+        type: sensor_code,
+      },
+    };
+
+    body.metadata = {
+      ...body.metadata,
+      ...sourceInfo,
+    };
 
-  const weightSamples: Sample[] = [];
-  tenoviWeight.forEach(reading => {
-    weightSamples.push({
-      time: reading.timestamp,
-      value: formatNumber(convert(getFloatValue(reading.value_1)).from("lb").to("kg")),
+    const weightSamples: Sample[] = [];
+    tenoviWeight.forEach(reading => {
+      weightSamples.push({
+        time: reading.timestamp,
+        value: formatNumber(convert(getFloatValue(reading.value_1)).from("lb").to("kg")),
```

Comment:
NABD, but to help reading/maintaining this we could create some local constants with intermediate steps to get the final `value`

Reply:
Yea def good practice to do that!